### PR TITLE
Dockerfile: removed specifics, added php.ini overrides

### PR DESCRIPTION
**Pull request by bryanlatten**
Also added php.ini overrides in the form of user.ini

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -31,16 +31,16 @@ RUN locale-gen en_US.UTF-8 && export LANG=en_US.UTF-8 && add-apt-repository ppa:
...
         php5-memcached=2.2.0-2+deb.sury.org~trusty+1 \
-        php5-mcrypt=5.6.10+dfsg-1+deb.sury.org~trusty+1 \
-        php5-json=1.3.6-1+deb.sury.org~trusty+2 \
+        php5-dev \
+        php5-gd \
+        php5-mysqlnd \
+        php5-intl \
+        php5-curl \
+        php5-mcrypt \
+        php5-json \
```
Comment (bryanlatten) [inline Dockerfile:43]: these were unnecessarily strict and typically tied only to major releases of PHP

File: Dockerfile [hunk 2]
```diff
@@ -52,19 +52,9 @@ RUN printf "\n" | pecl install apcu-4.0.7 && \
...
 
-
 # Enable apache rewrite module
 RUN a2enmod rewrite
 
-# Install New Relic monitoring tools
-# IMPORTANT: will not work without setting license key (nrsysmond-config --set license_key=YOUR_LICENSE_KEY)
-# @see https://docs.newrelic.com/docs/release-notes/agent-release-notes/php-release-notes/php-agent-418089
-#RUN sh -c 'echo deb http://apt.newrelic.com/debian/ newrelic non-free >> /etc/apt/sources.list.d/newrelic.list' && \
-#    wget -O- https://download.newrelic.com/548C16BF.gpg | apt-key add - && \
```
Comment (bryanlatten) [inline Dockerfile:63]: not going to be used, removed.

File: Dockerfile [hunk 3]
```diff
@@ -31,16 +31,16 @@ RUN locale-gen en_US.UTF-8 && export LANG=en_US.UTF-8 && add-apt-repository ppa:
...
     apt-get -yq install \
         php5=5.6.10+dfsg-1+deb.sury.org~trusty+1 \
-        php5-dev=5.6.10+dfsg-1+deb.sury.org~trusty+1 \
-        php5-gd=5.6.10+dfsg-1+deb.sury.org~trusty+1 \
-        php5-mysqlnd=5.6.10+dfsg-1+deb.sury.org~trusty+1 \
-        php5-intl=5.6.10+dfsg-1+deb.sury.org~trusty+1 \
-        php5-curl=5.6.10+dfsg-1+deb.sury.org~trusty+1 \
         php5-gearman=1.1.2-1+deb.sury.org~trusty+2 \
         php5-memcache=3.0.8-5+deb.sury.org~trusty+1 \
         php5-memcached=2.2.0-2+deb.sury.org~trusty+1 \
```
Comment (bryanlatten) [inline Dockerfile:36]: these continue to be strictly versioned, as they represent functionality changes for those extensions

File: Dockerfile [hunk 4]
```diff
@@ -31,16 +31,16 @@ RUN locale-gen en_US.UTF-8 && export LANG=en_US.UTF-8 && add-apt-repository ppa:
...
-        php5-mcrypt=5.6.10+dfsg-1+deb.sury.org~trusty+1 \
-        php5-json=1.3.6-1+deb.sury.org~trusty+2 \
+        php5-dev \
```
Comment (mannytoledo) [inline Dockerfile:37]: Do we actually need php5-dev on these?

File: container/root/etc/php5/apache2/conf.d/overrides.user.ini [hunk 5]
```diff
@@ -0,0 +1,9 @@
+; Use this file to override specific defaults in the stock Ubuntu php.ini
+; NOTE: this only applies to web requests through Apache, not CLI (worker,cron) instantiations
+
+memory_limit = 192M
```
Comment (mannytoledo) [inline container/root/etc/php5/apache2/conf.d/overrides.user.ini:4]: Yay! More memory!

**Comment by mannytoledo** [comment]
One comment, otherwise :+1:

**Comment by jimdelois** [comment]
:white_check_mark:

**Comment by bossjones** [comment]
:+1:

File: Dockerfile (hunk 4 above)
Comment (bryanlatten) [inline Dockerfile:37]: @mannytoledo the idea here is...downstream apps can easily install extensions, there is a cleanup script that the same downstream app can run that then removes it (see https://github.com/bryanlatten/docker-php/blob/master/container/root/clean.sh)

**Pull request merged**